README.md

```diff
@@ -849,183 +976,179 @@ platformOptions.preSpawnProcessConfigurator = { _, fileAttr in
-        public var closeAllUnknownFileDescriptors: Bool
-        /// An ordered list of steps in order to tear down the child
-        /// process in case the parent task is cancelled before
-        /// the child proces terminates.
-        /// Always ends in sending a `.kill` signal at the end.
-        public var teardownSequence: [TeardownStep] = []
-        /// A closure to configure platform-specific
-        /// spawning constructs. This closure enables direct
-        /// configuration or override of underlying platform-specific
-        /// spawn settings that `Subprocess` utilizes internally,
-        /// in cases where Subprocess does not provide higher-level
-        /// APIs for such modifications.
-        ///
-        /// On Linux, Subprocess uses `fork/exec` as the
-        /// underlying spawning mechanism. This closure is called
-        /// after `fork()` but before `exec()`. You may use it to
-        /// call any necessary process setup functions.
-        public var preSpawnProcessConfigurator: (
-            @convention(c) @Sendable () -> Void
-        )? = nil
-
-        public init() {}
-    }
+/// The collection of Linux specific configurations
+@available(FoundationPreview 0.4, *)
+@available(iOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+public struct PlatformOptions: Sendable, Hashable {
+    // Set user ID for the subprocess
+    public var userID: uid_t?
+    /// Set the real and effective group ID and the saved
+    /// set-group-ID of the subprocess, equivalent to calling
+    /// `setgid()` on the child process.
+    /// Group ID is used to control permissions, particularly
+    /// for file access.
+    public var groupID: gid_t?
+    // Set list of supplementary group IDs for the subprocess
+    public var supplementaryGroups: [gid_t]?
+    /// Set the process group for the subprocess, equivalent to
+    /// calling `setpgid()` on the child process.
+    /// Process group ID is used to group related processes for
+    /// controlling signals.
+    public var processGroupID: pid_t?
+    // Creates a session and sets the process group ID
+    // i.e. Detach from the terminal.
+    public var createSession: Bool
+    // Whether the subprocess should close all file
+    // descriptors except for the ones explicitly passed
+    // as `input`, `output`, or `error` when `run` is executed
+    // This is equivelent to setting `POSIX_SPAWN_CLOEXEC_DEFAULT`
+    // on Darwin. This property is default to be `false`
+    // because `POSIX_SPAWN_CLOEXEC_DEFAULT` is a darwin-specific
+    // extension and we can only emulate it on Linux.
+    public var closeAllUnknownFileDescriptors: Bool
+    /// An ordered list of steps in order to tear down the child
+    /// process in case the parent task is cancelled before
+    /// the child proces terminates.
+    /// Always ends in sending a `.kill` signal at the end.
+    public var teardownSequence: [TeardownStep] = []
```

Comment:
I guess this is where these are used, but how do you create them other than the one static func?